tests/resources/test_iam_role.py

```diff
@@ -39,6 +61,10 @@ def test_policies(iam_role):
     assert policies[0].PolicyName == "root"
 
 
+def test_all_conditions(iam_role):
+    assert len(iam_role.all_statement_conditions) == 2
```

Comment:
I'd add here more asserts, maybe check that the conditions returned are the same as the ones in the template?
Do you agree?